sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/caches/AsyncCache.java

```diff
@@ -74,15 +75,15 @@ public void set(TKey key, TValue value) {
                 }
 
                 logger.debug("cache[{}] result value is obsolete ({}), computing new value", key, obsoleteValue);
-                AsyncLazy<TValue> asyncLazy = new AsyncLazy<>(singleValueInitFunc);
+                AsyncLazy<TValue> asyncLazy = new AsyncLazy<>(singleValueInitFunc, Optional.of(value));
```

Comment:
There are two scenarios 

- Current value (don't block on future refreshed) { When request get started }
- Current value if different from previous value 
     Current value if different return immediately
     If not then wait for refresh

Assuming so isn't modeling current value into the future refresh hacky?

Reply:
I think these are already covered in the new added unit tests I added.